### PR TITLE
Allow adding digits and symbols in XkPasswd generator using mask-like values

### PR DESCRIPTION
**Pull request by glowinthedark**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description

- update xkpass fields to allow appending numbers+symbols via a mask such as `ddds` which would generate `random.password123!`
currently position of `d` and `s` is not currently considered; only the count is relevant

## :bulb: Motivation and Context

Some sites have requirements for passwords to contain numbers and special symbols. Introducing a field where you can specify `ds` would append one digit and one symbol correspondingly. `dddsss` would append 3 numbers and 2 special characters.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [x] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :camera_flash: Screenshots / GIFs

Screen Shot 2020-08-16 at 14 22 40

## After code review changes (fixed alignment for caps type and digits/symbols)
image

### CODE REVIEW

File: app/src/main/res/layout/fragment_xkpwgen.xml [hunk 1]
```diff
@@ -75,7 +75,29 @@
...
+            app:layout_constraintHorizontal_weight="0.5"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/separator" />
+            app:layout_constraintTop_toBottomOf="@id/total_words" />
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/xk_numbers_symbols_label"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
```
Comment (msfjarvis) [inline app/src/main/res/layout/fragment_xkpwgen.xml]: this needs a `marginTop` like the `Spinner`

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file.
 ### Added
 - Allow changing the branch used for Git operations
 - Allow setting a subdirectory key when creating folders
+- XKpasswd: add field for appending digits/symbols to the password by mask (e.g. `dds`)
```
Comment (msfjarvis) [inline CHANGELOG.md]:
```suggestion
- Allow adding digits/symbols in XkPasswd generated passwords using a mask-like value (`dds` gives you two digits and a symbol, and so on)
```

This reads a bit better

File: app/src/main/res/layout/fragment_xkpwgen.xml (hunk 1 above)
Comment (Skrilltrax) [inline app/src/main/res/layout/fragment_xkpwgen.xml]: Align the start and the end of the `@id/xk_numbers_symbols_label` with the `@id/separator` and remove `marginStart`. Also, align the `Spinner` with `@id/xkPasswordText`.

Comment (glowinthedark) [reply app/src/main/res/layout/fragment_xkpwgen.xml]: @Skrilltrax: for `xk_numbers_symbols_label` `start` can be aligned to `@id/separator` but the end in my understanding should be `app:layout_constraintEnd_toEndOf="parent"`, no? i.e. the same value as the field above.

```xml
<com.google.android.material.textfield.TextInputLayout
    android:id="@+id/xk_numbers_symbols_label"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintStart_toStartOf="@id/separator"
    app:layout_constraintTop_toBottomOf="@id/separator"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintTop_toTopOf="@id/xkCapType"
```

and you actually meant to align the Spinner with `@+id/total_words`, no? (not with `@id/xkPasswordText` which is higher up).

Also, since the alignment is now based on the widgets above I have removed `app:layout_constraintHorizontal_weight="0.5"` on both the spinner and `xk_numbers_symbols_label`.

I think I got it right now; please verify.

Moreover, originally I was testing in landscape orientation, but when switching to portrait I realised that with the new alignment the default label for the spinner `Sentencecase` was getting clipped, therefore I renamed both the label and the enum (because the code is getting the enum using `CapsType.valueOf(string)`).

Attached a new screenshot in the description after the alignment changes.

Comment (Skrilltrax) [reply app/src/main/res/layout/fragment_xkpwgen.xml]: Hey @glowinthedark you're right, I meant `@+id/total_words` and not `@id/xkPasswordText`. For `xk_numbers_symbols_label` end can also be set to `@id/separator`, doing this will make sure that if we increase or decrease the start or end margin of `@id/separator` in future, then `xk_numbers_symbols_label` will automatically align with it.

Comment (Skrilltrax) [reply app/src/main/res/layout/fragment_xkpwgen.xml]: A small nit, can you set `app:layout_constraintBottom_toBottomOf="@id/xk_numbers_symbols_label"` to the `Spinner`. That should center align `Spinner` with `xk_numbers_symbols_label`

File: app/src/main/java/com/zeapo/pwdstore/pwgenxkpwd/CapsType.kt [hunk 3]
```diff
@@ -5,5 +5,5 @@
 package com.zeapo.pwdstore.pwgenxkpwd
 
 enum class CapsType {
-    lowercase, UPPERCASE, TitleCase, Sentencecase, As_iS
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/pwgenxkpwd/CapsType.kt:8]: why this change?

Comment (glowinthedark) [reply app/src/main/java/com/zeapo/pwdstore/pwgenxkpwd/CapsType.kt:8]: @msfjarvis:  because the labels are the same as values (in the Spinner), and the enum value is taken as eg `TitleCase.valueOf("Sentencecase")`. The string `Sentencecase` was too long and was getting clipped in portrait mode therefore the enum had to be changed together with the string value.

As can be seen in the screenshot (in the description) `Sentencecase` was renamed to `Sentence`.

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/pwgenxkpwd/CapsType.kt:8]: Alright that makes sense

**Review by msfjarvis — APPROVED** [review]
